backend/app/anti_recommendation_engine/anti_recommendation_engine.py

```diff
@@ -23,41 +24,54 @@ class AntiRecommendationEngine:
     """
 
     def __init__(self) -> None:
+        self.__anti_recommender: AntiRecommender | None = None
+        self.__current_anti_recommendation_records: list[Record] = []
         self.__records_by_key: dict[RecordKey, Record] = {
             record.key: record for record in AllSourceReader().read()
         }
-        self.__anti_recommender: AntiRecommender | None = (
-            self.__select_anti_recommender()
-        )
         self.__stack: list[list[Record]] = []
-        self.__current_anti_recommendation_records: list[Record] = []
 
-    def __select_anti_recommender(self) -> AntiRecommender | None:
-        """Select and return an AntiRecommender based on values stored in settings."""
+    @staticmethod
+    def select_anti_recommender(
+        *, user_state: UserState, record_keys: tuple[RecordKey, ...]
+    ) -> AntiRecommender | None:
```

Comment:
I'd have a default implementation here that picked randomly, for example.

Reply:
`ArkgAntiRecommender` is now the default anti-recommender.